### PR TITLE
Support for Rails 5.1

### PR DESCRIPTION
**Pull request by Bramjetten**
There was a change in a private method called `build_relation` in Rails 5.1. I added an updated `uniqueness_validator.rb` in `lib/patches`.

### CODE REVIEW

File: lib/globalize/active_record/act_macro.rb [hunk 1]
```diff
@@ -40,15 +40,18 @@ def allow_translation_of_attributes(attr_names)
...
 
-        if ::ActiveRecord::VERSION::STRING > "5.0" && connected? && table_exists? && translation_class.table_exists?
-          self.ignored_columns += translated_attribute_names.map(&:to_s)
-          reset_column_information
+        begin
+          if ::ActiveRecord::VERSION::STRING > "5.0" && table_exists? &&translation_class.table_exists?
+            self.ignored_columns += translated_attribute_names.map(&:to_s)
+            reset_column_information
+          end
+        rescue ::ActiveRecord::NoDatabaseError
```
Comment (parndt) [inline lib/globalize/active_record/act_macro.rb]: Would you consider emitting a warning here?

File: lib/patches/active_record/rails5_1/uniqueness_validator.rb [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            relation = relation.where(find_finder_class_for(record).table_name => { scope_item => scope_value })
+          end
+
+          translated_scopes.each do |scope_item|
+            scope_value = record.send(scope_item)
+            relation = relation.where(scope_item => scope_value)
+          end
+          relation = relation.merge(options[:conditions]) if options[:conditions]
+
+          # if klass.unscoped.with_translations.where(relation).exists?
```
Comment (parndt) [inline lib/patches/active_record/rails5_1/uniqueness_validator.rb]: Did you mean to leave this commented out line in?

File: lib/globalize/active_record/act_macro.rb [hunk 3]
```diff
@@ -57,6 +60,7 @@ def check_columns!(attr_names)
              "Attribute name(s): #{overlap.join(', ')}\n"].join
           )
         end
+      rescue ::ActiveRecord::NoDatabaseError
```
Comment (parndt) [inline lib/globalize/active_record/act_macro.rb]: Same as above my thinking is we should warn the user

File: lib/patches/active_record/rails5_1/uniqueness_validator.rb (hunk 2 above)
Comment (Bramjetten) [inline lib/patches/active_record/rails5_1/uniqueness_validator.rb]: This was simply copied from `rails5/uniqueness_validator.rb`. I removed it.

File: lib/globalize/active_record/act_macro.rb (hunk 1 above)
Comment (Bramjetten) [inline lib/globalize/active_record/act_macro.rb]: Added warnings.

**Comment by Bramjetten** [comment]
@parndt Is there anything that needs to get done before this can be merged? Happy to help.

**Comment by raonibr** [comment]
+1, I'm also waiting for it.

**Comment by waldyr** [comment]
+1

**Comment by kevin-jj** [comment]
@parndt @dalpo   It would be great to merge this PR

**Comment by Lax** [comment]
👍 +1 waiting for it

**Pull request merged**